From 67f9041cffd0d8865a29f96c52e2fbeb69523346 Mon Sep 17 00:00:00 2001
From: 24bit192khz <118996408+24bit192kHz@users.noreply.github.com>
Date: Sun, 4 Oct 2026 15:05:35 +0300
Subject: [PATCH 24/28] [local] payloads/GRUB2: pack the site-local signing key
 and password into the memdisk

site-local/data/boot.pub and auth.cfg, when present, go to /boot/grub in
the GRUB memdisk next to the runtime config, so they are part of the
payload coreboot measures into PCR 2.

---
 payloads/external/GRUB2/Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/payloads/external/GRUB2/Makefile b/payloads/external/GRUB2/Makefile
index 01fff7d37c..d1c4633d83 100644
--- a/payloads/external/GRUB2/Makefile
+++ b/payloads/external/GRUB2/Makefile
@@ -49,6 +49,10 @@ EXTRA_PAYLOAD_FILES=/boot/grub/site.cfg=$(abspath ../../../$(RUNTIME_CFG))
 else
 EXTRA_PAYLOAD_FILES=/boot/grub/site.cfg=$(RUNTIME_CFG)
 endif
+ifeq ($(CONFIG_GRUB2_INCLUDE_RUNTIME_CONFIG_FILE),y)
+# site-local boot policy: signing key and GRUB password (auth.cfg is not in git)
+EXTRA_PAYLOAD_FILES+=$(foreach f,boot.pub auth.cfg,$(if $(wildcard ../../../site-local/data/$(f)),/boot/grub/$(f)=$(abspath ../../../site-local/data/$(f))))
+endif
 
 grub2: config
 	echo "    MAKE       GRUB2 $(NAME-y)"
